contrib/offline/upload2artifactory.py

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python3
+""" recursively upload each file to private repo in Artifactory """
+""" Uses environment variables: USERNAME, TOKEN, and BASE_URL. """
+import os
+import subprocess
+
+
+def upload_files(base_url, username, token):
+    """recurse current dir and upload each file with curl"""
+    for root, dirs, files in os.walk(os.getcwd()):
+        for file in files:
+            file_path = os.path.join(root, file)
+            relative_path = os.path.relpath(file_path, os.getcwd())
+            destination_url = f"{base_url}/{relative_path}"
+
+            # Construct and execute the curl command
+            command = [
+                "curl",
+                "-u",
+                f"{username}:{token}",
+                "-T",
+                file_path,
+                destination_url,
+            ]
+            print(f"Uploading {file_path} to {destination_url}")
+            try:
+                result = subprocess.run(
+                    command, check=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+                )
+                print(f"Success: {result.stdout.decode('utf-8')}")
+            except subprocess.CalledProcessError as e:
+                print(f"Error uploading {file_path}: {e.stderr.decode('utf-8')}")
```

Comment:
This is contrib/, so I don't care that much, but why shell out ? Since you're using python, might as well do http calls in it ?

Reply:
Fair point.